packages/lesswrong/components/posts/PostsPage/PostsPageWrapper.tsx

```diff
@@ -3,33 +3,64 @@ import { Components, registerComponent } from '../../../lib/vulcan-lib';
 import { useSingle, UseSingleProps } from '../../../lib/crud/withSingle';
 import { isMissingDocumentError, isOperationNotAllowedError } from '../../../lib/utils/errorUtil';
 import { isPostWithForeignId } from "./PostsPageCrosspostWrapper";
+import { commentGetDefaultView } from '../../../lib/collections/comments/helpers';
+import { useCurrentUser } from '../../common/withUser';
+import { useMulti } from '../../../lib/crud/withMulti';
+import { viewNames } from '../../comments/CommentsViews';
+import { useSubscribedLocation } from '../../../lib/routeUtil';
+import { isServer } from '../../../lib/executionEnvironment';
 
 const PostsPageWrapper = ({ sequenceId, version, documentId }: {
   sequenceId: string|null,
   version?: string,
   documentId: string,
 }) => {
+  const currentUser = useCurrentUser();
+  const { query } = useSubscribedLocation();
+
+  const extraVariables = {sequenceId: 'String', ...(version && {version: 'String'}) }
+  // Note: including batchKey ensures that the post query is not batched together with the
+  // comments query when sent from the client (i.e. not during SSR). This makes the main post body load
+  // much faster
+  const extraVariablesValues = {sequenceId, batchKey: "singlePost", ...(version && {version}) }
+  const fragmentName = version ? 'PostsWithNavigationAndRevision' : 'PostsWithNavigation'
+
   const fetchProps: UseSingleProps<"PostsWithNavigation"|"PostsWithNavigationAndRevision"> = {
     collectionName: "Posts",
-    ...(version ? {
-      fragmentName: 'PostsWithNavigationAndRevision',
-      extraVariables: {
-        version: 'String',
-        sequenceId: 'String',
-      },
-      extraVariablesValues: { version, sequenceId },
-    } : {
-      fragmentName: 'PostsWithNavigation',
-      extraVariables: {
-        sequenceId: 'String',
-      },
-      extraVariablesValues: { sequenceId },
-    }),
+    fragmentName,
+    extraVariables,
+    extraVariablesValues,
     documentId,
   };
 
   const { document: post, refetch, loading, error } = useSingle<"PostsWithNavigation"|"PostsWithNavigationAndRevision">(fetchProps);
 
+  // This section is a performance optimisation to make comment fetching start as soon as possible rather than waiting for
+  // the post to be fetched first. This is mainly beneficial in SSR
+
+  // Note: in principle defaultView can depend on the post (via post.commentSortOrder). In practice this is almost never set,
+  // less than 1/1000 posts have it set. If it is set the consequences are that the comments will be fetched twice. This shouldn't
+  // cause any rerendering or significant performance cost (relative to only fetching them once) because the second fetch doesn't wait
+  // for the first to finish.
```

Comment:
*This is the important caveat. Normally I wouldn't consider doing this optimistic fetching thing to be worth it because of this, but in this case I think it is because:
 - commentSortOrder being set is so rare
 - the posts page gets a lot of traffic, so there is more benefit to small performance optimisations